mlflow/tensorflow/__init__.py

```diff
@@ -1340,7 +1341,9 @@ def __restore_generator(prev_generator):
                 history=history,
             )
             # Ensure all data are logged.
-            mlflow.flush_async_logging()
+            # Shut down the async logging (instead of flushing)
+            # to avoid leaving zombie threads between patchings.
+            _shut_down_async_logging()
```

Comment:
@chenmoneygithub Curious what happens if we don't set up threads. Does the cause an issue?

Reply:
I think if we don't activate the async_logging but we use log_batch_async then there will be an issue